flink-connector-mysql-cdc/src/main/java/com/ververica/cdc/connectors/mysql/source/enumerator/MySqlSourceEnumerator.java

```diff
@@ -85,12 +87,17 @@ public MySqlSourceEnumerator(
         this.sourceConfig = sourceConfig;
         this.splitAssigner = splitAssigner;
         this.readersAwaitingSplit = new TreeSet<>();
+        if (splitAssigner.getAssignerStatus() == NEWLY_ADDED_ASSIGNING
```

Comment:
We can use `AssignerStatus.isAssigning(splitAssigner) || AssignerStatus.isAssigningFinished(splitAssigner)` to cover newly assigning and initial assigning.